### ISSUE TITLE
image with width=1 or height=1

### ISSUE THREAD

**Issue by JackC92** [alecjacobson/computer-graphics-raster-images#63]
I saw a previous issue about this, but I don't think there's anything wrong with it. Is there any convention that a valid image should have at least two rows and two columns?

**Comment by amacle** [comment]
Nope! A 1px by 1px image is perfectly valid

**Comment by lvlingxiao1** [comment]
Then for demosaic, if there is only one row, we might not have any red/blue to average, what should the value be then? Should it be zero?

**Comment by amacle** [comment]
yes, the image would not be tall enough to encode/store any red channel values when applying the Bayer filter as-described, so the red values when demosaicing would be zero
